tests/functionaltests/functionaltest-api.h

```diff
@@ -385,4 +402,88 @@
 #endif
 #define EXPECT_OBJCNE_MSG(val1, val2, format, ...)                                   \
     FunctionalTestLog::SetUserComment(GTestLogPrivate::Format(format, __VA_ARGS__)); \
-    EXPECT_OBJCNE(val1, val2);
\ No newline at end of file
+    EXPECT_OBJCNE(val1, val2);
+
+// clang-format off
+#pragma push_macro("TEST")
+#define TEST(className, methodName)  \
+    \
+    class className##_##methodName { \
+        TEST_CLASS(className##_##methodName) \
+    \
+        TEST_METHOD(Test); \
+    };  \
+    \
+    void className##_##methodName::Test()
+
+#pragma push_macro("TEST_F")
+#define TEST_F(className, methodName)                  \
+/* This assumes that there already exists a className type with SetUp and TearDown methods to call */ \
+    \
+    class className##_##methodName : className { \
```

Comment:
>className##_##methodName [](start = 10, length = 24)

assumed to be unique. Also means that we don't get nice grouping in the output but I'll take what we can get here. An alternative would be to add new macros that do nothing in gtest land but make a new test class in taef land. the normal TEST macros could then just become TEST_METHOD. This would let us do fancier things with test class properties but would require modifying every test file and firmly put us in a franken test framework state that is neither GTEST or TAEF.